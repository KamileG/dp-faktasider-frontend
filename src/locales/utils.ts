import { useContext } from 'react';
import { LocaleContext } from './LocaleContext';

export type SupportedLanguage = 'en' | 'nn' | 'nb';

export const useLocale = () => {
  return useContext(LocaleContext).lang;
};
