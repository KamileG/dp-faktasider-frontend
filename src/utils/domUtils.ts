export function getDomElemnt(id: string) {
  return document.getElementById(id);
}

export function pxFromTop(id: string): number {
  return getDomElemnt(id)?.getBoundingClientRect().top || Number.POSITIVE_INFINITY;
}
