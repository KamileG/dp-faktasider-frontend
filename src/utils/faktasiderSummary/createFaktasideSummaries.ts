import { Translations } from '../../types/translations';
import { SupportedLanguage, supportedLanguages } from '../../i18n/supportedLanguages';
import localizeSanityContent from '../../i18n/localizeSanityContent';
import { Oppsett, RawFaktasideData } from '../../../gatsby-utils/createFaktasider';

export interface RawFaktasideSummary {
  title?: Translations<string>;
  ingress?: Translations<string>;
  visSprakversjon?: Translations<boolean | null>;
  id: string;
  slug?: {
    current: string;
  };
}

export interface FaktasideSummary {
  path: string;
  tittel: string;
  språk: SupportedLanguage;
  tilgjengeligPåValgtSpråk: boolean;
  ingress: string;
  id: string;
}

function createFaktasideSummaries(
  faktasider: RawFaktasideData[],
  oppsett: Oppsett,
  lang: SupportedLanguage
): FaktasideSummary[] {
  const sorteringsMal: string[] = oppsett.faktasideSortering.map((it) => it.id);
  const sortedPages: RawFaktasideSummary[] = sorteringsMal.map(
    (id) => faktasider.find((page) => page.id === id) as RawFaktasideSummary
  );
  const unsortedPages = faktasider.filter((page) => !sorteringsMal.includes(page.id));

  return [...sortedPages, ...unsortedPages].map((page) => createFaktasideSummary(page, lang));
}

export function createFaktasideSummary(page: RawFaktasideSummary, lang: 'en' | 'no') {
  const slug = page.slug?.current;
  const oversettelser = supportedLanguages.filter((lang) => page.visSprakversjon?.[lang]);
  const tilgjengeligPåValgtSpråk = oversettelser.includes(lang);
  const språk = tilgjengeligPåValgtSpråk ? lang : oversettelser[0];
  const path = `/${språk}/${slug}/`;
  const tittel = localizeSanityContent(page.title, lang) as string;
  const ingress = localizeSanityContent(page.ingress, lang) as string;

  return {
    path,
    tittel,
    språk,
    tilgjengeligPåValgtSpråk,
    ingress,
    id: page.id,
  };
}

export default createFaktasideSummaries;
