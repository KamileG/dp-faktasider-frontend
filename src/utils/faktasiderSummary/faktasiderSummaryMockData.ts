import { FaktasideSummary } from './useFaktasiderSumary';

export const faktasiderSummaryMockData: FaktasideSummary[] = [
  {
    path: '/no/arbeidsledig/',
    tittel: 'Arbeidsledig',
    språk: 'no',
    tilgjengeligPåValgtSpråk: true,
    ingress: 'Har du blitt arbeidsledig, kan du ha rett til økonomisk støtte og hjelp til å komme i arbeid.',
    id: '-89eddf21-6b78-5f89-8d1f-7f5f8ebfe735',
  },
  {
    path: '/no/permittert/',
    tittel: 'Permittert',
    språk: 'no',
    tilgjengeligPåValgtSpråk: false,
    ingress: 'Er du permittert, kan du ha rett til økomisk støtte og hjelp til å komme i arbeid.',
    id: '-21aed679-d1fa-5238-ac15-c39b39ce3011',
  },
];
