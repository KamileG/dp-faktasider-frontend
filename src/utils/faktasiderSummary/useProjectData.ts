import { graphql, useStaticQuery } from 'gatsby';
import { useLocale } from '../../i18n/LocaleContext';
import { Translations } from '../../types/translations';
import localizeSanityContent from '../../i18n/localizeSanityContent';

interface Data {
  oppsett: {
    title: Translations<string>;
  };
}

export interface ProjectData {
  title: string;
}

function useProjectData(): ProjectData {
  const lang = useLocale();
  const data: Data = useStaticQuery(graphql`
    query Oppsett {
      oppsett: sanityOppsett {
        title: _rawTitle
      }
    }
  `);

  const localizedData = localizeSanityContent(data, lang);

  return {
    title: localizedData.oppsett.title,
  };
}

export default useProjectData;
