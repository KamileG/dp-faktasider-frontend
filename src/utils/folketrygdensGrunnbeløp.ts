export function useGrunnbellop() {
  const G = 99858;

  return {
    GtoNOK: (g: number) => Math.round(g * G).toLocaleString('nb-NO'),
    G: G,
  };
}
