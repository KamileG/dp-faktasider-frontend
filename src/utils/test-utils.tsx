import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { TranslationsProvider } from '../i18n/i18nextConfig';
import { VisForContextProvider } from '../components/BlockContent/VisFor/VisForContext';
import { DevContextProvider } from '../components/DevKnapper/DevContext';
import { LocaleProvider } from '../i18n/LocaleContext';

// https://testing-library.com/docs/react-testing-library/setup#custom-render
const AllTheProviders = ({ children }) => (
  <LocaleProvider lang="no">
    <VisForContextProvider>
      <DevContextProvider>
        <TranslationsProvider>{children}</TranslationsProvider>
      </DevContextProvider>
    </VisForContextProvider>
  </LocaleProvider>
);

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'queries'>) =>
  // @ts-ignore
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
