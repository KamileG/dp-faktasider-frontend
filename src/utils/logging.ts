/*
Vi må lazy-loade loggingConfig, da gatsby pre-rendrer html, og kresjer dermed amplitude-sdk
Prøver derfor å ikke initiere amplitude-sdk instansen før en logevent blir kalt.
*/

import { isProduction } from './environment';

const loggEvent = (event: string, ekstraData?: object) => {
  if (!isProduction()) {
    return;
  }

  import('../utils/loggingConfig').then((logging) =>
    logging.loggInstance.logEvent(event, {
      ...ekstraData,
      appName: 'dp-faktasider',
    })
  );
};

export const loggError = (error: Error, ekstraData?: object) =>
  loggEvent('Error', {
    ...ekstraData,
    siteUrl: window.location.pathname,
    msg: error.message,
    name: error.name,
    stack: error.stack,
  });

export const loggRedirect = (fraLenke: string) => loggEvent('Redirect fra gammel lenke', { fraLenke });

export const loggSidevisning = (side: string) => loggEvent('Side vist', { side });

export const loggIkkeOversatt = (side: string) => loggEvent('Ikke oversatt', { side });

export const loggKalkulatorbruk = (type: string) => loggEvent('Brukt kalkulator', { type });

export const loggHashlenkeKlikk = () => loggEvent('Brukt hashlenke');

export const loggMeny = (type: string) => loggEvent('Meny', { type });

export const loggH2MenyKlikk = () => loggEvent('H2Meny', { type: 'Hopp til underoverskrift' });

export const loggVisTilleggsinfo = (tittel: string) => loggEvent('Vis tilleggsinfo', { tittel });

export const loggTilpassInnhold = (situasjon: string) => loggEvent('Tilpasset innhold', { situasjon });

export const loggNotFound = (path: string) => loggEvent('404 - not found', { path });

export const loggHashNotFound = (hash: string, path: string) => loggEvent('Hashlenke finnes ikke', { path, hash });
