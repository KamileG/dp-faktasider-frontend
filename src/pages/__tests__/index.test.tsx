import React from 'react';
import { PureIndexPage } from '../index';
import { faktasiderSummaryMockData } from '../../utils/faktasiderSummary/faktasiderSummaryMockData';
import { render } from '../../utils/test-utils';

const indexStaticQueryMockData = {
  oppsett: {
    title: {
      _type: 'localeString',
      no: 'Arbeid',
    },
  },
};

test('Snapshottest landingsside', () => {
  const { container } = render(
    <PureIndexPage projectData={indexStaticQueryMockData} sider={faktasiderSummaryMockData} />
  );
  expect(container).toMatchSnapshot();
});
