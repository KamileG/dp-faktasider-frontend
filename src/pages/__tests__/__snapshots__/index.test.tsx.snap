// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshottest landingsside 1`] = `
.c0 {
  background-color: #CCE1F3;
  border-bottom: 0.3rem solid #99C2E8;
  border-top: 0.3rem solid #99C2E8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c2 {
  font-size: 1.7rem;
}

.c1 {
  padding: 2rem;
  max-width: 50rem;
}

.c5 {
  background-color: white;
  padding: 1.2rem 1.2rem 2rem;
  border-radius: 0.5rem;
}

.c3 {
  max-width: 50rem;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2rem 0;
}

.c3 .c4 {
  -webkit-flex: 15rem 1;
  -ms-flex: 15rem 1;
  flex: 15rem 1;
  margin: 0.5rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 1rem;
  font-size: 1.2rem;
}

.c7 {
  text-align: center;
  opacity: 0.7;
  margin: 1rem 0 !important;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="typo-sidetittel c2"
      >
        Arbeid
      </h1>
      <p
        class="typo-normal"
      />
    </div>
  </div>
  <div
    class="c3"
  >
    <div
      class="c4 c5"
    >
      <a
        class="c6"
        href="/no/arbeidsledig/"
      >
        Arbeidsledig
      </a>
      <p>
        Har du blitt arbeidsledig, kan du ha rett til økonomisk støtte og hjelp til å komme i arbeid.
      </p>
    </div>
    <div
      class="c4 c5"
    >
      <a
        class="c6"
        href="/no/permittert/"
      >
        Permittert
      </a>
      <p
        class="c7"
      >
        (
        Kun tilgjengelig på
         
        Norsk
        )
      </p>
      <p>
        Er du permittert, kan du ha rett til økomisk støtte og hjelp til å komme i arbeid.
      </p>
    </div>
  </div>
</div>
`;
